### PR TITLE
Update ProjectModel server

### PR DESCRIPTION
**Pull request by troydai**
1. Update version of dependnecy item;
2. Update framework name in dependency message
3. Fix dependency message regression
4. Update tests

Fixing issuse: https://github.com/dotnet/cli/issues/802. https://github.com/dotnet/cli/issues/805

/cc @davidfowl

### CODE REVIEW

**Comment by troydai** [comment]
Ping

File: src/Microsoft.DotNet.ProjectModel.Server/Helpers/NuGetFrameworkExtensions.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
```
Comment (troydai) [inline src/Microsoft.DotNet.ProjectModel.Server/Helpers/NuGetFrameworkExtensions.cs]: Ignore the changes in this file. I'll revert them.

File: src/Microsoft.DotNet.ProjectModel.Server/Helpers/NuGetFrameworkExtensions.cs [hunk 2]
```diff
@@ -10,13 +11,28 @@ public static class NuGetFrameworkExtensions
...
         public static FrameworkData ToPayload(this NuGetFramework framework)
         {
+            var frameworkName = new FrameworkName(framework.DotNetFrameworkName);
+
             return new FrameworkData
             {
                 ShortName = framework.GetShortFolderName(),
                 FrameworkName = framework.DotNetFrameworkName,
-                FriendlyName = framework.Framework,
+                FriendlyName = $"{framework.Framework} {framework.GetNormalizedVersionString()}",
```
Comment (davidfowl) [inline src/Microsoft.DotNet.ProjectModel.Server/Helpers/NuGetFrameworkExtensions.cs]: https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Dnx.DesignTimeHost/ProjectStateResolver.cs#L99

Comment (troydai) [inline src/Microsoft.DotNet.ProjectModel.Server/Helpers/NuGetFrameworkExtensions.cs]: Working on it.

**Comment by troydai** [comment]
Ping @davidfowl

**Comment by troydai** [comment]
Ping @davidfowl @anurse

File: src/Microsoft.DotNet.ProjectModel.Server/InternalModels/ProjectContextSnapshot.cs [hunk 3]
```diff
@@ -24,31 +25,38 @@ internal class ProjectContextSnapshot
...
+            var allExports = context.CreateExporter(configuration).GetAllExports();
             var allSourceFiles = new List<string>(context.ProjectFile.Files.SourceFiles);
             var allFileReferences = new List<string>();
             var allProjectReferences = new List<ProjectReferenceDescription>();
             var allDependencies = new Dictionary<string, DependencyDescription>();
-            
-            foreach (var export in context.CreateExporter(configuration).GetDependencies())
+
+            var frameworkExports = allExports.Where(export => export.Library.Identity.Type == LibraryType.ReferenceAssembly)
+                                             .ToDictionary(export => export.Library.Identity.Name);
```
Comment (anurse) [inline src/Microsoft.DotNet.ProjectModel.Server/InternalModels/ProjectContextSnapshot.cs]: Looks like you always end up having to strip off the `fx/` when looking things up in this dictionary. How about you apply the prefix to the keys here so you don't have to keep doing that?

File: src/Microsoft.DotNet.ProjectModel.Server/Helpers/LibraryExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.ProjectModel.Graph;
+
+namespace Microsoft.DotNet.ProjectModel.Server.Helpers
+{
+    public static class LibraryExtensions
+    {
+        public static string GetNameForDictionary(this LibraryDescription library)
```
Comment (troydai) [inline src/Microsoft.DotNet.ProjectModel.Server/Helpers/LibraryExtensions.cs]: not sure about the name. advice?

**Pull request merged**